### ISSUE TITLE
util.ipyDecode returned msg.header.msg_type mismatch with shell_router

### ISSUE THREAD

**Issue by Vesnica** [facebook/iTorch#61]
Fresh installed torch follow by: http://torch.ch/docs/getting-started.html#_

When I start a new itorch notebook, console shows:
```
qlua: /home/vesnica/torch/install/share/lua/5.1/itorch/main.lua:313: Cannot find appropriate message handler for comm_open
stack traceback:
        [C]: at 0x7fde03c70970
        [C]: in function 'assert'
        /home/vesnica/torch/install/share/lua/5.1/itorch/main.lua:313: in function </home/vesnica/torch/install/share/lua/5.1/itorch/main.lua:311>
        /home/vesnica/torch/install/share/lua/5.1/lzmq/poller.lua:75: in function 'poll'
        ...e/vesnica/torch/install/share/lua/5.1/lzmq/impl/loop.lua:307: in function 'poll'
        ...e/vesnica/torch/install/share/lua/5.1/lzmq/impl/loop.lua:325: in function 'sleep_ex'
        ...e/vesnica/torch/install/share/lua/5.1/lzmq/impl/loop.lua:370: in function 'start'
        /home/vesnica/torch/install/share/lua/5.1/itorch/main.lua:344: in main chunk
        [C]: in function 'require'
        [string "arg={'/run/user/1000/jupyter/kernel-80af0bdf-..."]:1: in main chunk
```
I look at the code(main.lua):
```
local function handleShell(sock)
   local msg = util.ipyDecode(sock)
   assert(shell_router[msg.header.msg_type],
          'Cannot find appropriate message handler for ' .. msg.header.msg_type)
   return shell_router[msg.header.msg_type](sock, msg);
end
```
Apparently **msg.header.msg_type == comm_open** and **shell_router[comm_open] == nil**

And I check the code, shell_router only have following members:
* connect_request
* kernel_info_request
* shutdown_request
* execute_request
* history_request
* complete_request
* object_info_request

My ipython version is 4.0.0, notebook version is 4.0.1

**Comment by soumith** [comment]
hmm, this seems new to ipython 4.0+. I will implement compability for this version when I get time. For now, you can simply downgrade your ipython version to 2.4.xx

**Comment by Vesnica** [comment]
yeah, I read the latest ipython messaging protocol docs, there is no easy fix for it.

Anyway, thanks for the hard work.

**Comment by soumith** [comment]
was solved via #64
